docker/jenkins/Makefile

```diff
@@ -2,13 +2,22 @@ BASE_FILE=docker-compose.yml
 OVERRIDE_FILE=docker-compose.override.yml
 SERVICE=ci
 SETUP_SCRIPT=../nvidia/setup_nvidia_docker_compose.py
+ENV_VARIABLES=CUDA_VERSION GPU_ARCH
 
 all: setup_nvidia_docker
 
 setup_nvidia_docker: ${OVERRIDE_FILE}
 
-${OVERRIDE_FILE}: ${BASE_FILE} ${SETUP_SCRIPT} Makefile
+${OVERRIDE_FILE}: ${BASE_FILE} ${SETUP_SCRIPT} Makefile ${ENV_VARIABLES}
 	${SETUP_SCRIPT} --extended-file ${BASE_FILE} --extend-services ${SERVICE} --override-file ${OVERRIDE_FILE}
 
 clean:
-	rm -vf *_generated ${OVERRIDE_FILE}
+	rm -vf *_generated ${OVERRIDE_FILE} ${ENV_VARIABLES}
+
+.PHONY: always-rebuild
+
+.SILENT: ${ENV_VARIABLES}
+${ENV_VARIABLES}: always-rebuild
```

Comment:
what does this do? just curious

Reply:
Every time we call `make setup_nvidia_docker` this rule will be executed and check whether the environment variables we care about changed.